pyteal/ast/abi/array_static.py

```diff
@@ -78,6 +78,21 @@ def __init__(self, array_type_spec: StaticArrayTypeSpec[T, N]) -> None:
     def type_spec(self) -> StaticArrayTypeSpec[T, N]:
         return cast(StaticArrayTypeSpec[T, N], super().type_spec())
 
+    def decode(
+        self,
+        encoded: Expr,
+        *,
+        startIndex: Expr = None,
+        endIndex: Expr = None,
+        length: Expr = None,
+    ) -> Expr:
+        if endIndex is not None or length is not None:
+            raise TealInputError(
+                "Expected None for endIndex and length on static array"
+            )
+
+        return super().decode(encoded, startIndex=startIndex, length=self.byte_length())
```

Comment:
nit: let's consider a case where static array has elements that are dynamic, then this static byte length trick cannot work, so I propose we should add a condition check here for dynamic array element.

Reply:
I added a check and a couple tests in the DYNAMIC_TYPES list in array_base_test.py

Reply:
seems correct to me, resolving